### PR TITLE
Update the SPCR table to version 4 and added RISC-V debug subtype to DBG2

### PR DESCRIPTION
**Pull request by david-sia**
Updated the SPCR Table to ver 4 to support:
- RISC-V SBI
- Precise Baud Rate
- NamespaceStringLength
- NamespaceStringOffset
- NamespaceString

Additionally, RISC-V debug subtype is added to DBG2 header.

### CODE REVIEW

**Comment by xiaobo55x** [comment]
LGTM!

File: source/compiler/dttemplate.h [hunk 1]
```diff
@@ -1896,7 +1896,7 @@ const unsigned char TemplateSlit[] =
 
 const unsigned char TemplateSpcr[] =
 {
-    0x53,0x50,0x43,0x52,0x50,0x00,0x00,0x00,  /* 00000000    "SPCRP..." */
+    0x53,0x50,0x43,0x52,0x5A,0x00,0x00,0x00,  /* 00000000    "SPCRZ..." */
     0x01,0xE3,0x49,0x4E,0x54,0x45,0x4C,0x20,  /* 00000008    "..INTEL " */
```
Comment (vlsunil) [inline source/compiler/dttemplate.h:1900]: I think you need to update the revision and checksum as well.

Comment (david-sia) [reply source/compiler/dttemplate.h:1900]: noted. will do

Comment (sacdintel) [reply source/compiler/dttemplate.h:1900]: Thanks @vlsunil for the review, I will take a look as well next week once those suggestions have been taken, thanks!

Comment (sacdintel) [reply source/compiler/dttemplate.h:1900]: Please don't forget to update the template too once all the changes are made!

Comment (david-sia) [reply source/compiler/dttemplate.h]: Hi @xiaobo55x , @vlsunil , @sacdintel ,

I have updated the code. Please help to review again. Thanks.

**Review by vlsunil — APPROVED** [review]
Looks good to me.

Thanks,
Reviewed-by: Sunil V L <sunilvl@ventanamicro.com>

**Comment by sacdintel** [comment]
Thanks @vlsunil and @xiaobo55x for the review! LGTM now and thanks for rebasing it on top of the new release as well, great open source software contribution etiquettes!

**Pull request merged**